File: server/lib/logger.ts
import pino from "pino";

const isDev = process.env.NODE_ENV !== "production";

export const logger = pino({
  level: process.env.LOG_LEVEL || "info",
  transport: isDev
    ? {
        target: "pino-pretty",
        options: {
          colorize: true,
          translateTime: "SYS:yyyy-mm-dd HH:MM:ss.l",
          ignore: "pid,hostname",
        },
      }
    : undefined,
  base: {
    service: "intellihire-server",
    version: "1.0.0",
  },
});

export function logRequest(
  method: string,
  url: string,
  statusCode: number,
  durationMs: number,
  userId?: string
) {
  const logData = {
    method,
    url,
    statusCode,
    durationMs,
    ...(userId && { userId }),
  };

  if (statusCode >= 500) {
    logger.error(logData, "Server error response");
  } else if (statusCode >= 400) {
    logger.warn(logData, "Client error response");
  } else {
    logger.info(logData, "Request completed");
  }
}

File: server_py/app/logger.py
import os
import sys
import logging

LOG_LEVEL = os.getenv("LOG_LEVEL", "INFO").upper()
is_dev = os.getenv("NODE_ENV", "development") != "production"

# Configure logging format
log_format = "%(asctime)s - %(name)s - %(levelname)s - %(message)s"
if is_dev:
    log_format = "%(asctime)s - [%(levelname)s] - %(message)s"

logging.basicConfig(
    level=getattr(logging, LOG_LEVEL, logging.INFO),
    format=log_format,
    handlers=[logging.StreamHandler(sys.stdout)]
)

logger = logging.getLogger("intellihire-server")

def log_request(
    method: str,
    url: str,
    status_code: int,
    duration_ms: float,
    user_id: str = None
):
    log_data = f"method={method} url={url} status_code={status_code} duration_ms={duration_ms:.2f}ms"
    if user_id:
        log_data += f" user_id={user_id}"

    if status_code >= 500:
        logger.error(f"Server error response: {log_data}")
    elif status_code >= 400:
        logger.warning(f"Client error response: {log_data}")
    else:
        logger.info(f"Request completed: {log_data}")
